### PR TITLE
python312Packages.primp: init at 0.6.4

### LINKED ISSUE #349432: Build failure: pyreqwest-impersonate
### Build log

https://hydra.nixos.org/build/274261734

Looks like missing a NEEDED of libstdc++?

### Notify maintainers

@drupol 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by drupol**
Should fix: https://github.com/NixOS/nixpkgs/issues/349432

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/python-aliases.nix [hunk 1]
```diff
@@ -474,6 +474,7 @@ mapAliases ({
   pyres = throw "pyres has been removed, since it is abandoned and broken"; # added 2023-06-20
   pygeoip = throw "pygeoip has been removed, since it wash archived in April 2018"; # added 2024-07-27
   pygeos = throw "pygeos has been removed, since it was merged with shapely 2.0"; # added 2024-05-26
+  pyreqwest-impersonate = throw "pyreqwest has been renamed to primp"; # added 2024-10-22
```
Comment (fabaff) [inline pkgs/top-level/python-aliases.nix:477]: Looks good to me as it's not a drop-in replacement (`import primp` vs. ìmport pyreqwest_impersonate').

**Comment by drupol** [comment]
@NixOS/darwin-core Do you have a clue on how to fix this?

**Comment by reckenrode** [comment]
Try adding `xcrun` as a `nativeBuildInput`. The ofborg failures appears to be due to trying to invoke `xcrun` when building the boring-sys crate.

https://github.com/cloudflare/boring/blob/3044e9ba98fcbac852cb9df9ee699d8d6b9ae1fb/boring-sys/build/main.rs#L421-L424

However, the build failure reported in https://github.com/NixOS/nixpkgs/issues/349432 is on staging-next. If you can target staging-next, the Darwin stdenv includes `xcrun` and a full SDK. You can delete the Darwin-specific `buildInputs` in that case.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349614`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 10 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349614`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349614`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.duckduckgo-search</li>
    <li>python311Packages.duckduckgo-search.dist</li>
    <li>python311Packages.primp</li>
    <li>python311Packages.primp.dist</li>
    <li>python312Packages.duckduckgo-search</li>
    <li>python312Packages.duckduckgo-search.dist</li>
    <li>python312Packages.primp</li>
    <li>python312Packages.primp.dist</li>
  </ul>
</details>

**Comment by drupol** [comment]
@reckenrode Thank you, rebased it against `staging-next` ! :)

**Pull request merged**